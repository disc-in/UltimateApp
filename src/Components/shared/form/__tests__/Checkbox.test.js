import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { Formik } from 'formik';
import * as Yup from 'yup';
import { TouchableOpacity } from 'react-native';

import Checkbox from '../Checkbox';

describe('Checkbox', () => {
  const values = ['red', 'blue'];
  const labels = ['I prefer red', 'I prefer blue'];

  it('renders correctly', () => {
    const { toJSON } = render(
      <Formik initialValues={{ title: ['My title'] }}>
        <Checkbox fieldName="title" label="Title" labels={labels} values={values} />
      </Formik>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when there is an error', async () => {
    const { toJSON, getByTestId, getByText } = render(
      <Formik
        initialValues={{ title: ['blue'] }}
        validationSchema={Yup.object({ title: Yup.array(Yup.string().oneOf(['blue'], 'Invalid title')) })}
      >
        {({ handleSubmit }) => (
          <>
            <Checkbox fieldName="title" label="Title" labels={labels} values={values} />
            <TouchableOpacity onPress={handleSubmit} testID="submit" />
          </>
        )}
      </Formik>,
    );

    fireEvent.press(getByText('I prefer red'));

    await waitFor(() => {
      fireEvent.press(getByTestId('submit'));
    });

    expect(getByText('Invalid title')).not.toBeNull();
    expect(toJSON()).toMatchSnapshot();
  });
});
