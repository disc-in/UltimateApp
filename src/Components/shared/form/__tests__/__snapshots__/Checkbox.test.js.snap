// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <Text
    style={
      Object {
        "marginBottom": 5,
      }
    }
  >
    Title
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      accessibilityRole="checkbox"
      accessibilityState={
        Object {
          "checked": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#ededed",
          "borderRadius": 3,
          "borderWidth": 0,
          "flexBasis": "40%",
          "margin": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
          "padding": 0,
        }
      }
      testID="input-title-red"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "color": "#43484d",
              "fontWeight": "normal",
              "marginLeft": 10,
              "marginRight": 10,
            }
          }
          testID="checkboxTitle"
        >
          I prefer red
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="checkbox"
      accessibilityState={
        Object {
          "checked": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#ededed",
          "borderRadius": 3,
          "borderWidth": 0,
          "flexBasis": "40%",
          "margin": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
          "padding": 0,
        }
      }
      testID="input-title-blue"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "color": "#43484d",
              "fontWeight": "normal",
              "marginLeft": 10,
              "marginRight": 10,
            }
          }
          testID="checkboxTitle"
        >
          I prefer blue
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Checkbox renders correctly when there is an error 1`] = `
Array [
  <View
    style={
      Object {
        "marginBottom": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "marginBottom": 5,
        }
      }
    >
      Title
    </Text>
    <Text
      style={
        Object {
          "color": "red",
          "fontStyle": "italic",
        }
      }
    >
      Invalid title
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
    >
      <View
        accessibilityRole="checkbox"
        accessibilityState={
          Object {
            "checked": true,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#ededed",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexBasis": "40%",
            "margin": 5,
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 1,
            "padding": 0,
          }
        }
        testID="input-title-red"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#2a9e91",
                  "fontSize": 24,
                },
                Object {
                  "minWidth": 24,
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "color": "#43484d",
                "fontWeight": "normal",
                "marginLeft": 10,
                "marginRight": 10,
              }
            }
            testID="checkboxTitle"
          >
            I prefer red
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="checkbox"
        accessibilityState={
          Object {
            "checked": true,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#ededed",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexBasis": "40%",
            "margin": 5,
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 1,
            "padding": 0,
          }
        }
        testID="input-title-blue"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#2a9e91",
                  "fontSize": 24,
                },
                Object {
                  "minWidth": 24,
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "color": "#43484d",
                "fontWeight": "normal",
                "marginLeft": 10,
                "marginRight": 10,
              }
            }
            testID="checkboxTitle"
          >
            I prefer blue
          </Text>
        </View>
      </View>
    </View>
  </View>,
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="submit"
  />,
]
`;
